solver/src/solver/http_solver/buffers.rs

```diff
@@ -0,0 +1,66 @@
+use contracts::ERC20;
+use ethcontract::{batch::CallBatch, H160, U256};
+use futures::future::join_all;
+use shared::Web3;
+use std::collections::HashMap;
+
+const MAX_BATCH_SIZE: usize = 100;
+
+#[derive(Clone)]
+/// Computes the amount of "buffer" ERC20 balance that the http solver can use
+/// to offset possible rounding errors in computing the amounts in a solution.
+pub struct BufferRetriever {
+    web3: Web3,
+    settlement_contract: H160,
+}
+
+impl BufferRetriever {
+    pub fn new(web3: Web3, settlement_contract: H160) -> Self {
+        Self {
+            web3,
+            settlement_contract,
+        }
+    }
+}
+
+#[cfg_attr(test, mockall::automock)]
+#[async_trait::async_trait]
+pub trait BufferRetrieving: Send + Sync {
+    async fn get_buffers(&self, tokens: &[H160]) -> HashMap<H160, U256>;
```

Comment:
I was going to say I would prefer if this returned a result instead of skipping failing balances but I see now that it is possible that for a bad erc20 token to revert on balance retrieval.
We could still do the same check here as we do in some places (don't remember where from top of my head) that it only skips smart contract errors (Revert and InvalidOpcode) but bubbles up any other error.